src/Disks/ObjectStorages/AzureBlobStorage/AzureBlobStorageAuth.cpp

```diff
@@ -125,21 +126,22 @@ std::unique_ptr<BlobContainerClient> getAzureBlobContainerClient(
 
     auto blob_service_client = getAzureBlobStorageClientWithAuth<BlobServiceClient>(endpoint.storage_account_url, container_name, config, config_prefix);
 
-    if (!endpoint.container_already_exists.has_value())
+    try
     {
-        ListBlobContainersOptions blob_containers_list_options;
-        blob_containers_list_options.Prefix = container_name;
-        blob_containers_list_options.PageSizeHint = 1;
-        auto blob_containers = blob_service_client->ListBlobContainers().BlobContainers;
-        for (const auto & blob_container : blob_containers)
+        return std::make_unique<BlobContainerClient>(
+            blob_service_client->CreateBlobContainer(container_name).Value);
+    }
+    catch (const Azure::Storage::StorageException & e)
+    {
+        /// If container_already_exists is not set (in config), ignore already exists error.
+        /// (Conflict - The specified container already exists)
+        if (!endpoint.container_already_exists.has_value() && e.StatusCode == Azure::Core::Http::HttpStatusCode::Conflict)
         {
-            if (blob_container.Name == endpoint.container_name)
-                return getAzureBlobStorageClientWithAuth<BlobContainerClient>(final_url, container_name, config, config_prefix);
+            tryLogCurrentException("Container already exists, returning the existing container");
```

Comment:
@tavplubix yeah, the problem is here, before if `container_already_exists` is not set, it first check list of containers and only if it does not exists creates one, however this code suffers from TOCTOU (time of check - time of use), since between `ListBlobContainers` and `CreateBlobContainer` container can be created by another node.

So I've changed this to the following:
- if `container_already_exists=false` then it will not make any attempts to create container
- otherwise (`true` or not set) it will first create the container and then check the return status, if it is `container already exists` and `container_already_exists` is not set, then it will simply return existing container, otherwise will fail

I don't like this one extra error, possible solutions:
- check container existence anyway, but if not exists put create into try/catch
- or simply remove `tryLogCurrentException`